graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -26,20 +37,27 @@
 )
 
 
-def rename_schema(ast, renamings):
+def rename_schema(
+    ast: DocumentNode, renamings: Dict[str, Optional[str]]
+) -> RenamedSchemaDescriptor:
     """Create a RenamedSchemaDescriptor; types and query type fields are renamed using renamings.
 
     Any type, interface, enum, or fields of the root type/query type whose name
-    appears in renamings will be renamed to the corresponding value. Any such names that do not
-    appear in renamings will be unchanged. Scalars, directives, enum values, and fields not
-    belonging to the root/query type will never be renamed.
+    appears in renamings will be renamed to the corresponding value if the value is not None. If the
+    value is None, it will be suppressed in the renamed schema and queries will not be able to
+    access it.
+
+    Any such names that do not appear in renamings will be unchanged.
+
+    Scalars, directives, enum values, and fields not belonging to the root/query type will never be
+    renamed.
 
     Args:
         ast: Document, representing a valid schema that does not contain extensions, input
              object definitions, mutations, or subscriptions, whose fields of the query type share
              the same name as the types they query. Not modified by this function
-        renamings: Dict[str, str], mapping original type/field names to renamed type/field names.
-                   Type or query type field names that do not appear in the dict will be unchanged.
+        renamings: Dict[str, Optional[str]], mapping original type/root type field names to renamed
```

Comment:
Consider removing the types from the `Args` section. Now that we have typehints, adding the type here is redundant and could accidentally get out of date.

Reply:
Sounds good-- this PR is already pretty big so after this gets merged I'll be removing these in a separate PR